### PR TITLE
SAK-28070 The new link 'Move Thread(s)' should only appear to those user...

### PR DESCRIPTION
**Pull request by lorenamgUMU**
...s that have the right permissions.

### CODE REVIEW

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/UIPermissionsManagerImpl.java [hunk 1]
```diff
@@ -394,7 +394,9 @@ public boolean isMovePostings(DiscussionTopic topic, DiscussionForum forum)
       while (iter.hasNext())
       {
         DBMembershipItem item = (DBMembershipItem) iter.next();
-        if (item.getPermissionLevel().getMovePosting().booleanValue()
+        if ((item.getPermissionLevel().getMovePosting().booleanValue()
+            || item.getPermissionLevel().getReviseAny().booleanValue()
+            || item.getPermissionLevel().getReviseOwn().booleanValue())
```
Comment (buckett) [inline msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/UIPermissionsManagerImpl.java]: Why is being able to edit your own messages considered sufficient to be able to move a thread?

**Comment by lorenamgUMU** [comment]
Because I saw in the specification document of the "Move thread" jira (https://jira.sakaiproject.org/secure/attachment/26801/msgcntr-241_spec.pdf) that the user will be allowed to 'Move thread' if he has permissions to 'Edit thread'.
Permissions:
Site role with permission level that supports Edit Postings > All will be able to move any thread
• consequently will see checkbox next to every thread item and “Move Thread(s)” link on Topics page
• consequently will see “Move Thread” link on thread page for any thread within a topic
Site role with permission level that supports Edit Postings > Own will be able to move only a thread that
contains message(s) authored by user only
• consequently will only see checkbox and “Move Thread(s)” link next to only thread(s) authored by user
on Topics page
• consequently will see “Move Thread” link on thread page of which user is sole author
Site role with permission level that supports Edit Postings > None will not be able to move any thread
• consequently will not see any checkbox or “Move Thread(s)” link

**Comment by buckett** [comment]
Thanks for the clear explanation regarding the permissions.

**Comment by juanjmerono** [comment]
Travis-CI raised and error compiling this PR, but I think is not related with this change. In the other hand Travis-CI is showing "unknown authored and committed" in the build: https://travis-ci.org/sakaiproject/sakai/builds/49712297 , Could it be because @lorenamgUMU was not signed up in Travis-CI?

Anyway @buckett do you think is safe merge this PR?

**Comment by buckett** [comment]
Looks good to me, I didn't merge it because I don't know loads about msgcntr.
The build failure is because of a failed elasticsearch test: https://s3.amazonaws.com/archive.travis-ci.org/jobs/49712298/log.txt

**Comment by juanjmerono** [comment]
@buckett after merge this PR travis' build is broken :scream:, now is due to a test failure on SiteStats... it seems like travis' builds are not too much reliable :disappointed:

**Comment by buckett** [comment]
I think it's our tests that aren't too reliable in this case....

**Pull request merged**